task2_Gal_Malka/src/ChessPiece.cpp

```diff
@@ -0,0 +1,24 @@
+#include "ChessPiece.h"
+
+ChessPiece::ChessPiece(Color color, char rep, const Position& position, MaxDistance max) :
+	m_color(color), m_maxSquares(max), m_rep(rep), m_position(position) {};
+
+// Check if a piece can colide with another piece (or empty square)
+bool ChessPiece::canColide(const ChessPiece& other, const Direction d, const bool dest) const {
+	if (&other && (m_color == other.m_color || !dest))
```

Comment:
I don't think you need to implement this method in the abstract class `ChessPiece`. Just mark it as a pure virtual function